Source/Bogus/DataSets/Date.cs

```diff
@@ -144,31 +129,221 @@ public DateTimeOffset FutureOffset(int yearsToGoForward = 1, DateTimeOffset? ref
+      /// and ensures that the range indicators are in the correct order and both reference actual
+      /// <see cref="DateTime"/> values. This takes into account the fact that when Daylight Savings Time
+      /// comes into effect, there is a 1-hour interval in the local calendar which does not exist, and
+      /// <see cref="DateTime"/> values in this change are not meaningful.
+      /// 
+      /// This function only worries about the start and end times. Impossible <see cref="DateTime"/>
+      /// values within the range are excluded automatically by means of the <see cref="DateTime.ToLocalTime"/>
+      /// function.
+      /// 
+      /// The version of this function built targeting .NET Standard 1.3 does not check Daylight Savings Time
+      /// transitions, as this API does not expose Daylight Savings Time information.
+      /// </summary>
+      /// <param name="start">A ref <see cref="DateTime"/> to be adjusted forward out of an impossible date/time range if necessary.</param>
+      /// <param name="end">A ref <see cref="DateTime"/> to be adjusted backward out of an impossible date/time range if necessary.</param>
+      /// <param name="kind">A <see cref="DateTimeKind"/> indicating how the supplied <see cref="DateTime"/> values should be interpreted with respect to DST change windows.</param>
+      /// <param name="preferRangeBoundary">An out <see cref="DateTime"/> that indicates which value should be used if the supplied range is empty due to being entirely contained within a DST transition range.</param>
+      private void ComputeRealRange(ref DateTime start, ref DateTime end, DateTimeKind kind, out DateTime preferRangeBoundary)
+      {
+         preferRangeBoundary = end;
+
+         if (start > end)
+         {
+            var tmp = start;
+
+            start = end;
+            end = tmp;
+         }
+
+#if !NETSTANDARD1_3
+         if (kind == DateTimeKind.Local)
+         {
+            var window = GetForwardDSTTransitionWindow(start);
+
+            var startLocal = start.ToLocalTime();
+            var endLocal = end.ToLocalTime();
+
+            if ((startLocal >= window.Start) && (startLocal <= window.End))
+            {
+               start = new DateTime(window.Start.Ticks, start.Kind);
+
+               if (start == end)
+                  return;
+            }
+
+            window = GetForwardDSTTransitionWindow(end);
+
+            if ((end >= window.Start) && (end < window.End))
+            {
+               end = new DateTime(window.End.Ticks, end.Kind);
+
+               // We had to bump the end, meaning that the end was not already a valid
+               // DateTime value (within the DST transition range), so prefer the start
+               // instead.
+               preferRangeBoundary = start;
+            }
+
+            if (start > end)
+               throw new Exception("DateTime range does not contain any real DateTime values due to daylight savings transitions");
```

Comment:
If one is using `Faker` to generate `DateTime` values, it will already be avoiding DST windows, so the worst case should be `start == end`. You should never encounter a `start > end` due to DST windows. Or are you talking about potentially pre-existing tests that rely on undefined behaviour by passing in `start` values that are after `end` values and it used to "just work"? This should not be an issue because `ComputeRealRange` swaps out-of-order `start` and `end` before beginning its computations.